src/types/quote/cert_data.rs

```diff
@@ -0,0 +1,174 @@
+            return Err(anyhow!(
+                "cannot transform cert data into pck cert chain data"
+            ));
+        }
+
+        let cert_data_size = utils::read_from_bytes::<little_endian::U32>(&mut data)
+            .ok_or_else(|| anyhow!("incorrect buffer size"))?
+            .get();
+
+        let cert_data = utils::read_bytes(&mut data, cert_data_size as usize);
+        let cert_data = cert_data.strip_suffix(&[0]).unwrap_or(cert_data);
+
+        let pck_cert_chain = CertificateInner::load_pem_chain(cert_data)
+            .context("Failed to parse PCK certificate chain")?;
+
+        let pck_extension = pck_cert_chain
+            .first()
+            .context("CertChain")?
+            .tbs_certificate
+            .extensions
+            .as_ref()
+            .and_then(|extensions| {
+                extensions
+                    .iter()
+                    .find(|ext| SgxPckExtension::is_pck_ext(ext.extn_id.to_string()))
+            })
+            .ok_or_else(|| anyhow!("PCK Certificate does not contain a SGX Extension"))?;
+
+        Ok(QuotingEnclaveReportCertData {
+            qe_report,
+            qe_report_signature,
+            qe_auth_data,
+            pck_cert_chain: pck_cert_chain.clone(),
+            pck_extension: SgxPckExtension::from_der(pck_extension.extn_value.as_bytes())
+                .context("PCK Extension")?,
+        })
+    }
+
+    pub fn as_pck_cert_chain_data(&self) -> anyhow::Result<PckCertChainData> {
+        if self.cert_key_type.get() != CertificationKeyType::PckCertChain as u16 {
+            return Err(anyhow!(
+                "cannot transform cert data into pck cert chain data"
+            ));
+        }
+
+        let cert_data = self.cert_data.strip_suffix(&[0]).unwrap_or(self.cert_data);
+        let pck_cert_chain = CertificateInner::load_pem_chain(cert_data)
+            .context("Failed to parse PCK certificate chain")?;
+
+        let pck_extension = pck_cert_chain
+            .first()
+            .context("CertChain")?
+            .tbs_certificate
+            .extensions
+            .as_ref()
+            .and_then(|extensions| {
+                extensions
+                    .iter()
+                    .find(|ext| SgxPckExtension::is_pck_ext(ext.extn_id.to_string()))
+            })
+            .ok_or_else(|| anyhow!("PCK Certificate does not contain a SGX Extension"))?;
+
+        Ok(PckCertChainData {
+            pck_cert_chain: pck_cert_chain.clone(),
+            pck_extension: SgxPckExtension::from_der(pck_extension.extn_value.as_bytes())
+                .context("PCK Extension")?,
+        })
+    }
+}
+
+pub struct QuotingEnclaveReportCertData<'a> {
+    pub qe_report: EnclaveReportBody,
+
+    pub qe_report_signature: Signature,
+
+    pub qe_auth_data: &'a [u8],
+
+    pub pck_cert_chain: Vec<CertificateInner>,
+
+    pub pck_extension: SgxPckExtension,
+}
+
+pub struct PckCertChainData {
+    pub pck_cert_chain: Vec<CertificateInner>,
```

Comment:
maybe we could just store the certificate chain as `PckCertChainData` here?

Reply:
Do you mean not to store the SgxExtensions ?

Reply:
Oops I was actually referencing lines 155 and 157. Since they both can technically be represented as `PckCertChainData`, but I will leave that up to you.

Reply:
Aah thank you ! Nice catch ! Fixed it https://github.com/automata-network/dcap-rs/pull/15/commits/d36fade6191d7089ee40e99421112d0b3fcb32a9